### PR TITLE
SeaPearl cleanup and bug fix for RL value selection

### PR DESCRIPTION
**Pull request by 3rdCore**
This is my proposition for SeaPearl cleanup. This pull request goes hand in hand with another [pull request](https://github.com/corail-research/SeaPearlZoo/pull/5) on SeaPearlZoo. 

**Problem :** Some part of the source code were obsolete and no longer in the framework of current SeaPearl development. The RL based knapsack example was crashing and returning untrackable errors : `ERROR: BoundsError: attempt to access 328-element Vector{Int64} at index [0]`

**Key features :**
- removed /exemple dir and obsolete examples
- removed no longer supported Seapearl files: `explorer.jl` and `learned.jl`
- removed dependencies to removed files
- fixed `knapsack_generator.jl`, some `IntVarView` were considered as _branchable_
- fixed agent's _value / index of value_ confusion that led to a bug

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -4,13 +4,18 @@ authors = ["Ilan Coulon", "Félix Chalumeau"]
 version = "0.2.0"
 
 [deps]
+CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
 DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
 Distributions = "31c24e10-a181-5473-b8eb-7969acd0382f"
 Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
 Flux = "587475ba-b771-5e3f-ad9e-33799f191a9c"
+Gadfly = "c91e804a-d5a3-530f-b6f0-dfbca275c004"
```
Comment (ilancoulon) [inline Project.toml:12]: Do we really need this kind of dep for SeaPearl?

File: src/CP/valueselection/lh_utils.jl [hunk 2]
```diff
@@ -53,46 +53,52 @@ end
...
     legal_actions = lh.action_space[legal_actions_mask]
 
+    #get reward 
     reward = lh.reward.value
+    
     # Initialize reward for the next state: not compulsory with DefaultReward, but maybe useful in case the user forgets it
     lh.reward.value = 0
 
-    # synchronize state: we could delete env.state, we do not need it 
+    # synchronize state: Update the adjacency_matrix of the CPLayerGraph using the CPModel
```
Comment (ilancoulon) [inline src/CP/valueselection/lh_utils.jl:80]: To be precise, the sync_state does not currently do that. This matrix is extracted when calling to_arraybuffer. The only thing that is (currently) synced here is the chosen variable and the possible values (two last columns of the array).

File: src/experiment/launch_experiment.jl [hunk 3]
```diff
@@ -66,14 +68,14 @@ function launch_experiment!(
...
+                verbose && println(" vs Visited nodes with basic Heuristic n°$(j-1) : ", model.statistics.numberOfNodes)
             end
-
-            # bestsolutions[i, j] = model.objectiveBound + 1
+            bestsolutions[i, j] = model.objectiveBound + 1
             nodevisited[i, j] = model.statistics.numberOfNodes
 
+            #TODO understand what this line is doing
             if j == 2
                 set_postfix(iter, Delta=string(nodevisited[i, 1] - nodevisited[i, 2]))
```
Comment (ilancoulon) [inline src/experiment/launch_experiment.jl:80]: This line is to update the progress bar, to show the current difference between the basic heuristic and the learned one (in terms of number of nodes).

File: test/runtests.jl [hunk 4]
```diff
@@ -13,5 +13,5 @@ const RL = ReinforcementLearning
     include("experiment/experiment.jl")
 end
 
-revise_user = "You use Revise, you're efficient in your work, well done ;)"
-revise_user
```
Comment (ilancoulon) [inline test/runtests.jl:17]: @felixchalumeau 🤣

**Review by qcappart — APPROVED** [review]

**Pull request merged**